cli/module_registry.go

```diff
@@ -176,46 +174,154 @@ func UpdateModuleCommand(c *cli.Context) error {
+	}
+
+	client, err := NewAppClient(c)
+	if err != nil {
+		return err
+	}
+
+	manifestPath := defaultManifestFilename
+	if manifestPathArg != "" {
+		manifestPath = manifestPathArg
+	}
+	var moduleID ModuleID
+	// if the manifest cant be found
+	if _, err := os.Stat(manifestPath); err != nil {
+		// no manifest found.
+		if nameArg == "" || (publicNamespaceArg == "" && orgIDArg == "") {
+			return errors.Errorf("Unable to find %s. "+
+				"If you want to upload a version without a %s, you must supply a module name and namespace (or module name and orgid)\n\n",
+				defaultManifestFilename, defaultManifestFilename)
+		}
+		moduleID = ModuleID{Prefix: publicNamespaceArg, Name: nameArg}
+	} else {
+		// if we can find a manifest, use that
+		manifest, err := loadManifest(manifestPath)
+		if err != nil {
+			fmt.Fprintf(c.App.ErrWriter, "If you want to upload a version without a %s, "+
+				"you must supply a module name and namespace (or module name and orgid)\n\n", defaultManifestFilename)
+			return err
+		}
+
+		moduleID, err = parseModuleID(manifest.Name)
+		if err != nil {
+			return err
+		}
+		if nameArg != "" && nameArg != moduleID.Name {
+			// This is almost certainly a mistake we want to catch
+			return errors.Errorf("Module name %q was supplied via command line args but the %s has a module name of %q",
+				nameArg, defaultManifestFilename, moduleID.Name)
+		}
+	}
+	// TODO(zaporter) This logic is duplicated in update and upload and should be refactored
+	// TODO(APP-2230) Simplify this once we can pass org-id in ModuleID
+	if publicNamespaceArg != "" {
+		switch moduleID.Prefix {
+		case "":
+			moduleID.Prefix = publicNamespaceArg
+		case publicNamespaceArg:
+			// the meta.json manifest == public_namespace arg
+			fmt.Fprintf(c.App.Writer, "The module's %s already specifies a public namespace. Ignoring\n", defaultManifestFilename)
+		default:
+			// the meta.json manifest != public_namespace arg
+			// we may want to investigate a better way of handling this error case
+			// For now, it seems like bad UX to ignore this error
+			return errors.Errorf("the module's %s specifies a namespace of '%s'"+
+				" but a namespace of '%s' was provided in command line arguments",
+				defaultManifestFilename, moduleID.Prefix, publicNamespaceArg)
+		}
+	}
+
+	var orgID *string
+	if orgIDArg != "" {
+		if moduleID.Prefix == "" {
+			orgID = &orgIDArg
+		} else {
+			fmt.Fprintf(c.App.Writer, "A public namespace (%s) is specified in the config."+
+				" It is not necessary to provide an org_id\n", moduleID.Prefix)
+		}
+	}
+	if orgID == nil && moduleID.Prefix == "" {
+		return errors.Errorf("The module's namespace is not set in %s."+
+			" You must provide a public_namespace (if you have set one) or supply your org id", defaultManifestFilename)
+	}
+	// end duplicated logic
+
+	//nolint:gosec
+	file, err := os.Open(tarballPath)
+	if err != nil {
+		return err
+	}
+	// TODO(APP-2226) support .tar.xz
```

Comment:
We also at one point wanted to support auto-compression but that seems like something to add after we have a MVP... too many design decisions are required to get that working for now

Reply:
There has been some debate already. The trend is to not assume specifics like this (how and when to compress) in our code but have documentation that recommends what todo up to the point of uploading.